src/charm.py

```diff
@@ -1,104 +1,328 @@
+            logger.info("Scaling is not implemented for this charm")
+            return
+        try:
+            self._charm_config: CharmConfig = CharmConfig.from_charm(charm=self)  # type: ignore[no-redef]  # noqa: E501
+        except CharmConfigInvalidError as exc:
+            event.add_status(BlockedStatus(exc.msg))
+            return
+        if not self._relation_created(F1_RELATION_NAME):
+            event.add_status(BlockedStatus("Waiting for F1 relation to be created"))
+            logger.info("Waiting for F1 relation to be created")
+            return
+        if not self._container.can_connect():
+            event.add_status(WaitingStatus("Waiting for container to be ready"))
+            logger.info("Waiting for container to be ready")
+            return
+        if not _get_pod_ip():
+            event.add_status(WaitingStatus("Waiting for Pod IP address to be available"))
+            logger.info("Waiting for Pod IP address to be available")
+            return
+        if not self._k8s_privileged.is_patched(container_name=self._container_name):
+            event.add_status(WaitingStatus("Waiting for statefulset to be patched"))
+            logger.info("Waiting for statefulset to be patched")
+            return
+        self.unit.set_workload_version(self._get_workload_version())
+        if not self._container.exists(path=BASE_CONFIG_PATH):
+            event.add_status(WaitingStatus("Waiting for storage to be attached"))
+            logger.info("Waiting for storage to be attached")
+            return
+        if not self._f1_requirer.f1_ip_address or not self._f1_requirer.f1_port:
+            event.add_status(WaitingStatus("Waiting for F1 information"))
+            logger.info("Waiting for F1 information")
+            return
+        event.add_status(ActiveStatus())
+
+    def _configure(self, _) -> None:
+        try:
+            self._charm_config: CharmConfig = CharmConfig.from_charm(charm=self)  # type: ignore[no-redef]  # noqa: E501
+        except CharmConfigInvalidError:
+            return
+        if not self._relation_created(F1_RELATION_NAME):
+            return
+        if not self._container.can_connect():
+            return
+        if not _get_pod_ip():
+            return
+        if not self._container.exists(path=BASE_CONFIG_PATH):
+            return
+        if not self._f1_requirer.f1_ip_address or not self._f1_requirer.f1_port:
+            return
+
+        if not self._k8s_privileged.is_patched(container_name=self._container_name):
+            self._k8s_privileged.patch_statefulset(container_name=self._container_name)
+        du_config = self._generate_du_config()
+        if config_update_required := self._is_du_config_up_to_date(du_config):
+            self._write_config_file(content=du_config)
+        service_restart_required = config_update_required
+        self._configure_pebble(restart=service_restart_required)
```

Comment:
Why the second variable here?
```suggestion
        if service_restart_required := self._is_du_config_up_to_date(du_config):
            self._write_config_file(content=du_config)
        self._configure_pebble(restart=service_restart_required)
```